include/geometrycentral/surface/barycentric_vector.h

```diff
@@ -76,8 +76,15 @@ struct BarycentricVector {
   // Rotate the vector 90 degrees CCW within the face it belongs to.
   // This requires the geometry, since the meaning of "90 degrees" depends on the geometry of the triangle.
   BarycentricVector rotated90(IntrinsicGeometryInterface& geom) const; // return a new vector
+  // Rotate the vector the specified amount of radians CCW within the face it belongs to.
+  // This again requires the geometry.
+  BarycentricVector rotated(IntrinsicGeometryInterface& geom, double angle) const;
```

Comment:
Is there a reason for 'rotated' rather than 'rotate' here? In some other libraries, present-tense vs. past-tense is used to indicated modification in-place vs returning  a new vector, although I think geometry-central does not use that convention and just always returns a new vector.

In the `Vector2` class we use 'rotate' https://github.com/nmwsharp/geometry-central/blob/master/include/geometrycentral/utilities/vector2.h#L62, so I would prefer to stick with that tense if possible.

Reply:
Oh, I guess maybe `rotated90` already existed, so you're following that convention?

That probably have been `rotate90` 😅

Reply:
I was under the impression that past tense is used to indicate returning a new vector, while present tense indicates in-place modification, for example [Eigen's `normalize()` vs. `normalized()` convention](https://eigen.tuxfamily.org/dox/group__QuickRefPage.html):
![image](https://github.com/nmwsharp/geometry-central/assets/27327961/5d972af0-7b3f-4124-b482-59fe49af9cc3)
Admittedly my decision was solely based this convention, and maybe Eigen is the exception! I'm fine with changing the tense to match the rest of geometry-central, though.